khard/khard.py

```diff
@@ -296,47 +296,103 @@ def list_address_books(address_books):
     print(helpers.pretty_print(table))
 
 
-def list_contacts(vcard_list):
+def list_contacts(vcard_list, fields=[], parsable=False):
     selected_address_books = []
     for contact in vcard_list:
         if contact.address_book not in selected_address_books:
             selected_address_books.append(contact.address_book)
     table = []
     # table header
     if len(selected_address_books) == 1:
-        print("Address book: {}".format(selected_address_books[0]))
-        table_header = ["Index", "Name", "Phone", "E-Mail"]
+        if not parsable:
+            print("Address book: {}".format(selected_address_books[0]))
+        table_header = ["index", "name", "phone", "email"]
     else:
-        print("Address books: {}".format(', '.join(
-            [str(book) for book in selected_address_books])))
-        table_header = ["Index", "Name", "Phone", "E-Mail", "Address book"]
+        if not parsable:
+            print("Address books: {}".format(', '.join(
+                [str(book) for book in selected_address_books])))
+        table_header = ["index", "name", "phone", "email", "address_book"]
     if config.show_uids:
-        table_header.append("UID")
-        abook_collection = AddressBookCollection('short uids collection',
-                                                 selected_address_books)
+        table_header.append("uid_short")
+
+    if parsable:
+        # Legacy default header fields for parsable.
+        table_header = ["uid", "name", "address_book"]
+
+    if fields:
+        table_header = [x.lower() for x in fields]
```

Comment:
I would suggest to use `x.capitalize().replace('_', ' ')` instead. This yields the old title in most cases and is generally a bit nicer for humans that the internal name.

Reply:
97a6568